BitwardenShared/Core/Auth/Repositories/AuthRepository.swift

```diff
@@ -636,10 +637,10 @@ extension DefaultAuthRepository: AuthRepository {
         // Clear all user data.
         try await biometricsRepository.setBiometricUnlockKey(authKey: nil)
         try await keychainService.deleteItems(for: userId)
-        try await stateService.logoutAccount(userId: userId, userInitiated: userInitiated)
-
-        // Remove the user from the timeout service and their SDK client.
         await vaultTimeoutService.remove(userId: userId)
+
+        // Log the account out last.
+        try await stateService.logoutAccount(userId: userId, userInitiated: userInitiated)
```

Comment:
🤔 Are there any dangers on moving this, because before if `logoutAccount` throws then `vaultTimeoutService` wouldn't remove the user id and now it will. I believe it should be fine but wanted to raise it just in case as in the other spot.

Reply:
Maybe, I think it should be ok though. The downside of doing anything after `logoutAccount` is that you can't rely on the function for getting the active account, since `logoutAccount` changes the active account. And that's easy to miss unless you know that it's happening after `logoutAccount`. So the old behavior seems worse in that regard.

Reply:
Yes, totally agree on that point. It's much safer this way.